internal/state/db/statepsql/members.go

```diff
@@ -333,5 +333,31 @@ func (db *db) GetUserInGuildHasRole(ctx context.Context, guildID int64, roleID i
 		return false, xerrors.Errorf("exec select: %w", err)
 	}
 
+	return exists, nil
+}
+
+func (db *db) GetUserInGuildsHasRoles(ctx context.Context, guildIDs []int64, roleIDs []int64, userID int64) (bool, error) {
+	const q = `
+	SELECT EXISTS(
+		SELECT
+			1
+		FROM
+			members
+		WHERE
+			guild_id = ANY ($1) AND user_id = $3 AND 
+			EXISTS (
+				SELECT 1
+				FROM jsonb_array_elements_text(data->'roles') AS role
+				WHERE role::text = ANY ($2::text[])
```

Comment:
role is text type? or int type?

Reply:
hmm role is text type so I shouldn't need to do "::text", will try and test again